threads/test_thread_rand.c

```diff
@@ -0,0 +1,88 @@
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/threads.h>
+
+#include "unity_fixture.h"
+
+
+typedef struct {
+	uint16_t time;
+	uint16_t id;
+} threads_self_t;
+
+
+static int cthread = 10, fin = 0;
+
+
+static char stack[10 * 512];
+
+
+static void test_thread(void *arg)
+{
+	threads_self_t *self = (threads_self_t *)arg;
+	usleep(1000 * self->time);
+	__sync_fetch_and_add(&fin, 1);
```

Comment:
Please don't use function embedded into the GNU CC compiler. Please synchronize global counter using mutex defined in _common structure. I don't understand why we should rely on GNU CC primitives. In this particular test we are not aiming to write high-performance code minimizing context switching. Rather we would like to test  correctness of constructing and destroying threads.

Reply:
Changed global counter synchronization. Now using `mutexLock()` and `mutexUnlock()` with lock stored in `thread_common` structure.